src/axom/inlet/Inlet.hpp

```diff
@@ -249,13 +252,39 @@ class Inlet : public SchemaCreator
    */
   bool get(const std::string& name, std::string& value);
 
+  /*!
+   *****************************************************************************
+   * \brief Sets the associated DocWriter for the Inlet instance.
+   *
+   * Sets the associated DocWriter. If the DocWriter is already set, it will be
+   * replaced by the one that was most recently set.
+   *
+   * \param [in] writer A pointer to a DocWriter object
+   *
+   *****************************************************************************
+   */
+  void registerDocWriter(std::shared_ptr<DocWriter> writer);
+
+  /*!
+   *****************************************************************************
+   * \brief Writes input deck documentation.
+   *
+   * This writes the input deck's documentation through the registered DocWriter.
+   *
+   *****************************************************************************
+   */
+  void writeDocs();
+
   // TODO add update value functions
 private:
   axom::sidre::View* baseGet(const std::string& name);
 
   std::shared_ptr<Reader> m_reader;
   axom::sidre::Group* m_sidreRootGroup = nullptr;
```

Comment:
I realize this predates your PR, but the preferred `C++11` way to initialize member variables is:
```suggestion
  axom::sidre::Group* m_sidreRootGroup{nullptr};
```